src/Models/Nonhydrostatic2DModels/equations_moisture.jl

```diff
@@ -1,5 +1,230 @@
+        bottom = Operators.SetValue(Geometry.WVector(FT(0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0))),
+    )
+    interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    interp_f2c = Operators.InterpolateF2C()
+    # flux_correction_center = Operators.FluxCorrectionC2C(
+    #     bottom = Operators.Extrapolate(),
+    #     top = Operators.Extrapolate(),
+    # )
+
+    # auxiliary variables
+    uvw = @. Geometry.Covariant123Vector(uh) +
+       Geometry.Covariant123Vector(interp_f2c(w))
+
+    # hyperdiffusion
+    χq = @. dρq_tot = hwdiv(hgrad(ρq_tot / ρ))
+    Spaces.weighted_dss!(dρq_tot)
+    @. dρq_tot = -κ₄ * hwdiv(ρ * hgrad(χq))
+
+    # advection
+    @. dρq_tot -= hdiv(uvw * (ρq_tot))
+    @. dρq_tot -= vector_vdiv_f2c(w * interp_c2f(ρq_tot))
+    @. dρq_tot -= vector_vdiv_f2c(interp_c2f(uh * (ρq_tot)))
+
+    # direct stiffness summation
+    Spaces.weighted_dss!(dρq_tot)
+end
+
+@inline function rhs_moisture!(
+    dY,
+    Y,
+    Ya,
+    t,
+    p,
+    Φ,
+    K,
+    ::ConservativeForm,
+    ::EquilibriumMoisture,
+    ::NoPrecipitation,
+    params,
+    # hyperdiffusivity,
+    FT,
+)
+    # parameters
+    #κ₄ = FT(0) #hyperdiffusivity
+
+    # unpack needed variables
+    dρq_tot = dY.moisture.ρq_tot
+    ρ = Y.base.ρ
+    ρuh = Y.base.ρuh
+    ρw = Y.base.ρw
+    ρq_tot = Y.moisture.ρq_tot
+
+    # operators /w boundary conditions
+    hdiv = Operators.Divergence()
+    hwdiv = Operators.WeakDivergence()
+    hgrad = Operators.Gradient()
+    vector_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(FT(0))),
+        top = Operators.SetValue(Geometry.WVector(FT(0))),
+    )
+    interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    interp_f2c = Operators.InterpolateF2C()
+    # flux_correction_center = Operators.FluxCorrectionC2C(
+    #     bottom = Operators.Extrapolate(),
+    #     top = Operators.Extrapolate(),
+    # )
+
+    # auxiliary variables
+    uvw = @. Geometry.Covariant123Vector(ρuh / ρ) +
+       Geometry.Covariant123Vector(interp_f2c(ρw) / ρ)
+
+    # # hyperdiffusion
+    # χq = @. dρq_tot = hwdiv(hgrad(ρq_tot / ρ))
+    # Spaces.weighted_dss!(dρq_tot)
+    # @. dρq_tot = -κ₄ * hwdiv(ρ * hgrad(χq))
+    dρq_tot .= FT(0)
+
+    # advection
+    @. dρq_tot -= hdiv(uvw * (ρq_tot))
+    @. dρq_tot -= vector_vdiv_f2c(ρw / interp_c2f(ρ) * interp_c2f(ρq_tot))
+    @. dρq_tot -= vector_vdiv_f2c(interp_c2f(ρuh / ρ * (ρq_tot)))
+
+    # direct stiffness summation
+    Spaces.weighted_dss!(dρq_tot)
 end
 
-@inline rhs_moisture!(dY, Y, Ya, t, p, ::Dry, params, FT) = nothing
+
+@inline function rhs_moisture!(
+    dY,
+    Y,
+    Ya,
+    t,
+    p,
+    Φ,
+    K,
+    ::ConservativeForm,
+    ::EquilibriumMoisture,
+    ::PrecipitationRemoval,
+    params,
+    # hyperdiffusivity,
+    FT,
+)
```

Comment:
I know that it's not the scope in this PR, but 12 arguments is kind of a lot.

Reply:
lol yep, we have a cache planned out ;) very soon very soon